multicluster/controllers/multicluster/label_identity_controller.go

```diff
@@ -0,0 +1,327 @@
+	r.labelAddsInProgress.Insert(labelToAdd)
+	// Unlock the labelMutex to allow cache updates for other labels. Any worker operating on
+	// the currentLabel will be blocked as it is added into labelAddsInProgress.
+	r.labelMutex.Unlock()
+	defer r.labelMutex.Lock()
+
+	resExportName := getResourceExportNameForLabelIdentity(r.localClusterID, labelToAdd)
+	labelResExport := r.getLabelIdentityResourceExport(r.remoteCommonArea.GetNamespace(), resExportName, labelToAdd)
+	err := r.remoteCommonArea.Create(ctx, labelResExport, &client.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+// deleteLabelIdentityResExport deletes a ResourceExport for a stale label. The function should be
+// invoked with the labelMutex locked, and will return with labelMutex remain locked.
+func (r *LabelIdentityReconciler) deleteLabelIdentityResExport(ctx context.Context, labelToDelete string) error {
+	r.labelDeletesInProgress.Insert(labelToDelete)
+	// Unlock the labelMutex to allow cache updates for other labels. Any worker operating on
+	// the originalLabel will be blocked as it is added into labelDeletesInProgress.
+	r.labelMutex.Unlock()
+	defer r.labelMutex.Lock()
+
+	labelResExport := &mcsv1alpha1.ResourceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getResourceExportNameForLabelIdentity(r.localClusterID, labelToDelete),
+			Namespace: r.remoteCommonArea.GetNamespace(),
+		},
+	}
+	err := r.remoteCommonArea.Delete(ctx, labelResExport, &client.DeleteOptions{})
+	return client.IgnoreNotFound(err)
+}
+
+func (r *LabelIdentityReconciler) getLabelIdentityResourceExport(resExportNamespace, name string, normalizedLabel string) *mcsv1alpha1.ResourceExport {
+	return &mcsv1alpha1.ResourceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: resExportNamespace,
+			Labels: map[string]string{
+				common.SourceKind:      common.LabelIdentityKind,
+				common.SourceClusterID: r.localClusterID,
+			},
+		},
+		Spec: mcsv1alpha1.ResourceExportSpec{
+			ClusterID: r.localClusterID,
+			Kind:      common.LabelIdentityKind,
+			LabelIdentity: &mcsv1alpha1.LabelIdentityExport{
+				NormalizedLabel: normalizedLabel,
+			},
+		},
+	}
+}
+
+func getNormalizedLabel(nsLabels, podLabels map[string]string, ns string) string {
+	if _, ok := nsLabels[v1.LabelMetadataName]; !ok {
+		// NamespaceDefaultLabelName is supported from K8s v1.21. For K8s versions before v1.21,
+		// we append the Namespace name label to the Namespace label set, so that the exported
+		// label is guaranteed to have Namespace name information.
+		nsLabels[v1.LabelMetadataName] = ns
+	}
+	return "ns:" + labels.FormatLabels(nsLabels) + "&pod:" + labels.FormatLabels(podLabels)
+}
+
+// getResourceExportNameForLabelIdentity retrieves the ResourceExport name for exporting
+// label identities in that cluster.
+func getResourceExportNameForLabelIdentity(clusterID string, normalizedLabel string) string {
```

Comment:
Ditto, it would be better to keep the same function signature format like `updateLabelCache` if they are the same types.
```suggestion
func getResourceExportNameForLabelIdentity(clusterID, normalizedLabel string) string {
```